lua/entities/gmod_wire_expression2/core/quaternion.lua

```diff
@@ -527,33 +544,51 @@ end
 __e2setcost(13)
 
 --- Performs spherical linear interpolation between <q0> and <q1>. Returns <q0> for <t>=0, <q1> for <t>=1
+--- Derived from c++ source on https://en.wikipedia.org/wiki/Slerp
 e2function quaternion slerp(quaternion q0, quaternion q1, number t)
-	local dot = q0[1]*q1[1] + q0[2]*q1[2] + q0[3]*q1[3] + q0[4]*q1[4]
-	local q11
-	if dot<0 then
-		q11 = {-q1[1], -q1[2], -q1[3], -q1[4]}
-	else
-		q11 = { q1[1], q1[2], q1[3], q1[4] }  -- dunno if just q11 = q1 works
+	local dot = qDot(q0, q1)
+
+	if dot < 0 then
+		q1 = {-q1[1], -q1[2], -q1[3], -q1[4]}
+		dot = -dot
+	end
+
+	if dot > 0.9995 then
+		local lerped = {
+			q0[1] + t*(q1[1] - q0[1]),
+			q0[2] + t*(q1[2] - q0[2]),
+			q0[3] + t*(q1[3] - q0[3]),
+			q0[4] + t*(q1[4] - q0[4]),
+		}
+		qNormalize(lerped)
+		return lerped
 	end
 
-	local l = q0[1]*q0[1] + q0[2]*q0[2] + q0[3]*q0[3] + q0[4]*q0[4]
-	if l==0 then return { 0, 0, 0, 0 } end
-	local invq0 = { q0[1]/l, -q0[2]/l, -q0[3]/l, -q0[4]/l }
-	local logq = qlog(qmul(invq0,q11))
-	local q = qexp( { logq[1]*t, logq[2]*t, logq[3]*t, logq[4]*t } )
-	return qmul(q0,q)
+	local theta_0 = acos(dot)
+	local theta = theta_0*t
+	local sin_theta = sin(theta)
+	local sin_theta_0 = sin(theta_0)
+
+	local s0 = cos(theta) - dot * sin_theta / sin_theta_0
+	local s1 = sin_theta / sin_theta_0
+
+	local slerped = {
+		q0[1]*s0 + q1[1]*s1,
+		q0[2]*s0 + q1[2]*s1,
+		q0[3]*s0 + q1[3]*s1,
+		q0[4]*s0 + q1[4]*s1,
+	}
+	qNormalize(slerped)
+	return slerped
 end
 
 -- Performs Linear interpolation between <q0> and <q1>. Returns <q0> for <t>=0, <q1> for <t>=1
 e2function quaternion lerp(quaternion q0, quaternion q1, number t, number reduceTo360)
-    local t1 = 1 - t
-    local dot = q0[1]*q1[1] + q0[2]*q1[2] + q0[3]*q1[3] + q0[4]*q1[4]
-
-    if reduceTo360 and dot < 0 then
-        return ( qmul(q0, t1) + qmul(q1, -t) )
```

Comment:
yeah, adding 2 tables to one another made no sense... who wrote that shit, anyway?

Reply:
#852 and apparently I argued against having lerp at all

Reply:
https://github.com/wiremod/wire/pull/852#issuecomment-77683148
Noone addressed my argument back then:
Why have lerp in the first place?
It's not used in practice and might confuse someone into thinking that this is the normal lerp that you're supposed to use.

Since it was always broken, how about we just remove it altogether and add a (working :)) nlerp instead?